colossus-metrics/src/main/scala/colossus/metrics/collectors/Counter.scala

```diff
@@ -22,13 +57,63 @@ class Counter private[metrics](val address: MetricAddress)(implicit collection:
     val values = counters.snapshot(false, false)
     if (values.isEmpty) Map() else Map(address -> values)
   }
-    
+}
 
+//Dummy implementation of a counter, used when "enabled=false" is specified at creation
+class NopCounter private[metrics](val address : MetricAddress) extends Counter {
+  val empty : MetricMap = Map()
+  override def tick(interval: FiniteDuration): MetricMap = empty
+
+  override def increment(tags: TagMap, amount: MetricValue): Unit = {}
+
+  override def set(tags: TagMap, value: MetricValue): Unit = {}
+
+  override def get(tags: TagMap): MetricValue = -1
```

Comment:
This should probably either return 0 or be wrapped in an option, not entirely sure which though.